src/common/types/function.ts

```diff
@@ -621,14 +624,21 @@ export class FunctionInstance {
             if (type.type !== 'never' && item.type === 'Input') {
                 const { id } = item.input;
                 const assignedType = partialInputs(id);
-                if (assignedType) {
-                    const converted = definition.convertInput(id, assignedType);
+                if (assignedType.output) {
+                    const converted = definition.convertInput(id, assignedType.output);
                     const newType = assign(converted, type).assignedType;
                     if (newType.type === 'never') {
-                        inputErrors.push({ inputId: id, inputType: type, assignedType });
+                        inputErrors.push({
+                            inputId: id,
+                            inputType: type,
+                            assignedType: assignedType.output,
+                        });
                     }
                     type = newType;
                 }
+                if (assignedType.sequence) {
+                    inputLengths.set(id, assignedType.sequence);
```

Comment:
This is wrong. You just set the input seq type regardless of whether it's actually assignable. What if a node requires a seq of length 10, and it's assigned a seq of length 20?

However, that's not all. Fixing assignability aside, there's also a problem with assigning the seq type per input. That's wrong as well. E.g. if you have a collector with 2 iterated input mapping to the same iterator input info, then assigning any one of them narrows the type of the other.

As an example, imagine a Merge Sprite sheet node that takes an image input and an index input (to give users absolute control over the order of images). The Image and Index inputs would both need to be of the same length (so they would use the same iter input info internally).

And there's a third problem related to other seq properties. If we add more properties to the `Sequence` type (and I plan on doing that in the future), then it will be hard to define the type value of the `IterInput<x>` variable(s). 

Using the modified Merge Sprite sheet node from above: Assuming that the Image input is assigned `Sequence { length: 10, foo: true }` and the Index input is assigned `Sequnce { length: 10 | 20, foo: false }`, what should the type of `IterInput0` be? It can't be `Sequence { length: 10 | 20, foo: true | false }`, because lengths must be narrowed, not widened. But it also can't be `Sequence { length: 10, foo: never }`, because that would evaluate the `never` type, which implies that an assignability error, even though this assignment should be fine.

To summarize: this change brings 3 problems:
1. No assignability check.
2. No type narrowing for related iterated inputs.
3. No clear way to define the type of the iterator.